### ISSUE TITLE
Erroneous MD003 warning

### ISSUE THREAD

**Issue by zafields** [DavidAnson/markdownlint#52]
`MD003/header-style: Header style [Expected: setext; Actual: atx]`

I get this warning for `### Header Level 3`, even though your documentation says it is valid...

> The setext_with_atx and settext_with_atx_closed doc styles allow atx-style headers of level 3 or more in documents with setext style headers:

**Comment by DavidAnson** [comment]
Can you please provide a small bit of Markdown that demonstrates the problem?

**Comment by DavidAnson** [comment]
I have an idea for what's going on. By default, the `style` for `MD003` is `consistent` which means the rule looks at the first header in the file and expects everything else to match that. If you want the `settext_with_atx_closed` setting to be used, you have to explicitly set it in config.

For example:

```json
{
  "MD003": {
    "style": "setext_with_atx_closed"
  }
}
```

References:
- https://github.com/DavidAnson/markdownlint/blob/master/doc/Rules.md#md003---header-style
- https://github.com/DavidAnson/markdownlint#configuration

**Comment by DavidAnson** [comment]
Please let me know if the above does not resolve your issue.
